commercetools-sdk-base/src/test/java/io/sphere/sdk/queries/PagedQueryResultTest.java

```diff
@@ -38,12 +40,47 @@ public void firstOfManyResult() throws Exception {
 
     @Test
     public void middleOfManyResult() throws Exception {
-        final PagedQueryResult<Integer> queryResult = PagedQueryResult.of((long) PAGE_SIZE, TOTAL, listOfSize(PAGE_SIZE));
+        final PagedQueryResult<Integer> queryResult = PagedQueryResult.of( (long)PAGE_SIZE, TOTAL, listOfSize(PAGE_SIZE));
         assertThat(queryResult.isFirst()).isFalse();
         assertThat(queryResult.isLast()).isFalse();
     }
 
     @Test
+    public void pagesIndexingResults() throws Exception {
+        {
+            final PagedQueryResult<Integer> queryResult = PagedQueryResult.of(0L, 0l, listOfSize(0));
+            assertThat(queryResult.getPageIndex()).isEqualTo(0);
+            assertThat(queryResult.getTotalPages()).isEqualTo(0);
+
+        }
+        {
+            final PagedQueryResult<Integer> queryResult = PagedQueryResult.of(0L, 106l, listOfSize(25));
+            assertThat(queryResult.getPageIndex()).isEqualTo(0);
+            assertThat(queryResult.getTotalPages()).isEqualTo(5);
+        }
+        {
+            final PagedQueryResult<Integer> queryResult = PagedQueryResult.of(75L, 106l, listOfSize(25));
+            assertThat(queryResult.getPageIndex()).isEqualTo(3);
+            assertThat(queryResult.getTotalPages()).isEqualTo(5);
+        }
+
+        {
+            final PagedQueryResult<Integer> queryResult = PagedQueryResult.of(100L, 106l, listOfSize(25));
+            assertThat(queryResult.getPageIndex()).isEqualTo(4);
+            assertThat(queryResult.getTotalPages()).isEqualTo(5);
+        }
+    }
+
+
+
+
+    @Test
+    public void getPagesCountForEmptyResults() throws Exception {
+        final PagedQueryResult<Integer> queryResult = PagedQueryResult.of( 80L, TOTAL, listOfSize(0));
+        assertThat(queryResult.getTotalPages()).isEqualTo(0);
```

Comment:
So in here you are saying there are no pages when actually there are, we just can't say how many because in the response we have no information about the limit (a pity, actually, but nothing we can do about it).
But just because we can't know, we should not give a number that would be valid in another situation. That's why I suggested in another comment (coming from a conversation with @katmatt ) to return an `Optional` instead of a number, and leave it clear the situations in which we can't tell. This would apply for the page index method too.